File: src/main/hava/de/richargh/sandbox/htmx/thymeleafsandbox/GreetingController.java
package de.richargh.sandbox.htmx.thymeleafsandbox;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class GreetingController {

    @GetMapping("/greeting")
    public String greeting(
            @AuthenticationPrincipal UserDetails userDetails,
            @RequestParam(name="name", required=false, defaultValue="World") String name,
            Model model) {
        String userName = userDetails != null ? userDetails.getUsername() : "None";
        System.out.println("/greeting for user: "+userName);
        model.addAttribute("name", name);
        return "greeting";
    }

}

File: src/main/kotlin/de/richargh/sandbox/htmx/kotlinxhtml/greeting/GreetingController.kt
package de.richargh.sandbox.htmx.kotlinxhtml.greeting

import de.richargh.sandbox.htmx.kotlinxhtml.commons.html.Context
import de.richargh.sandbox.htmx.kotlinxhtml.commons.html.PageContext
import de.richargh.sandbox.htmx.kotlinxhtml.commons.html.Paths
import de.richargh.sandbox.htmx.kotlinxhtml.commons.html.html
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class GreetingController {

    @GetMapping(Paths.Greeting.INDEX)
    fun greeting(
            @Context ctx: PageContext) =
            html(makeGreetingPage(ctx))

}
